src/meet-calendar.ts

```diff
@@ -0,0 +1,688 @@
+
+  /**
+   * Adds location info.
+   * @abstract
+   * @param text
+   */
+  abstract addLocationText(text: string): void;
+}
+
+/**
+ * Represents the description of the event.
+ */
+abstract class DescriptionWrapper {
+  protected event: EventContainer;
+
+  constructor(event: EventContainer) {
+    this.event = event;
+  }
+  /**
+   * Updates the description and location field is not already updated.
+   */
+  update(location: LocationWrapper) {
+    let isDescriptionUpdated = false;
+
+    // checks whether description was updated.
+    if (this.element !== undefined) {
+      let descriptionContainsURL =
+        this.value &&
+        this.value.length >= 1 &&
+        this.value.indexOf(BASE_URL) !== -1;
+
+      if (descriptionContainsURL) {
+        isDescriptionUpdated = true;
+      } else {
+        // checks whether there is the generated name in the location
+        // input if there is a location
+        if (location.text?.indexOf(LOCATION_TEXT) !== -1) {
+          isDescriptionUpdated = true;
+        }
+      }
+    }
+
+    if (isDescriptionUpdated) {
+      // update button url of event has all the data
+      this.updateButtonURL();
+    } else {
+      // update button as event description has no meeting set
+      this.updateInitialButtonURL(location);
+    }
+  }
+
+  /**
+   * Creates meeting, filling all needed fields.
+   * @param isDescriptionUpdated - whether description was already updated,
+   * true when we are editing event.
+   * @param the location to use to fill the meeting URL
+   */
+  clickAddMeeting(isDescriptionUpdated: boolean, location: LocationWrapper) {
+    if (!isDescriptionUpdated) {
+      // Build the invitation content
+      this.addDescriptionText(this.getInviteText());
+      this.updateButtonURL();
+
+      if (location) location.addLocationText(BASE_URL + this.event.meetingId);
+
+      // and create popup the window to create the meeting room
+      createRoom(BASE_URL + this.event.meetingId);
+    } else {
+      this.updateButtonURL();
+    }
+  }
+
+  /**
+   * The description html element.
+   * @abstract
+   */
+  abstract get element(): JQuery<HTMLElement> | undefined;
+  /**
+   * The text value of the description of the event.
+   * @abstract
+   */
+  abstract get value(): string;
+
+  /**
+   * Adds description text to the existing text.
+   * @abstract
+   * @param text
+   */
+  abstract addDescriptionText(text: string): void;
+
+  /**
+   * Generates description text used for the invite.
+   * @param dialInID optional dial in id
+   * @returns {String}
+   */
+  getInviteText(): string {
+    return `Click the following link to join the meeting from your computer: ${BASE_URL}${this.event.meetingId}`;
+  }
+
+  /**
+   * Updates the initial button text and click handler when there is
+   * no meeting scheduled.
+   */
+  updateInitialButtonURL(location: LocationWrapper) {
+    var button = $("#jitsi_button a");
+    button.html("Add a " + LOCATION_TEXT);
```

Comment:
can you use `.text` or some other safe method (that doesn't potentially inject HTML) for this and other `.html` calls?

Reply:
done, in this case and the others where `.html` was being used. Each was just setting plain text.